cmd/controller/kube/server.go

```diff
@@ -58,8 +58,8 @@ func (s *Server) GetIPInfo(ctx context.Context, req *kubepb.GetIPInfoRequest) (*
 }
 
 func (s *Server) GetClusterInfo(ctx context.Context, req *kubepb.GetClusterInfoRequest) (*kubepb.GetClusterInfoResponse, error) {
-	info, err := s.client.GetClusterInfo()
-	if err != nil {
+	info, err := s.client.GetClusterInfo(ctx)
+	if err != nil || info == nil {
```

Comment:
This probably should be `len(info) == 0`

Reply:
info type is `var info *ClusterInfo`, can't use len(info)

Reply:
Ahh my bad haha. Nevermind then!